### DISCUSSION TITLE
api registered user.confirmed sets true while model default is false

### DISCUSSION THREAD

**Discussion by hkovacs** [strapi/strapi#6530] [Legacy Topics]
**Describe the bug**
I really thought this was addressed somewhere else but couldnt find an existing issue.

this is an issue when posting to `/auth/local/register` when i check the user record in the admin i find that confirmed is set to `true`. 

my user model is:
```
    "confirmed": {
      "type": "boolean",
      "default": false,
      "configurable": false
    },
```

**Steps to reproduce the behavior**
post this user object to `/auth/local/register`

```
        const user = {
            username: 'bobobobob',
            email: '1@1.cc',
            password: '2222222',
            firstName: "how",
            lastName: "kov",
        }
```

**Expected behavior**
given that the model user.confirmed default is `false`, i expect `false` to be saved when registered.

**System**
Node.js version: v10.16.3
NPM version: 6.9.0
Strapi version: Strapi v3.0.1
Database: mongo v4.0.12
Operating system: ubuntu 18.04

**Additional context**
Thank you!

**Comment by alexandrebodin** [comment]
Hello @hkovacs this is the intended behaviour :) To disable auto confirmation you cano enable the `email_confirmation` option.  

The default field won't be taken into account as the auto confirmation is built into the code and set before the entry is saved in DB. If you need to change this you will need to extend the register function for now.

**Comment by hkovacs** [comment]
> Hello @hkovacs this is the intended behaviour :) To disable auto confirmation you cano enable the `email_confirmation` option.
> 
> The default field won't be taken into account as the auto confirmation is built into the code and set before the entry is saved in DB. If you need to change this you will need to extend the register function for now.

the behavior you describe doesnt match when user is created in admin. this seems inconsistent.

**Comment by derrickmehaffy** [comment]
> > Hello @hkovacs this is the intended behaviour :) To disable auto confirmation you cano enable the `email_confirmation` option.
> > The default field won't be taken into account as the auto confirmation is built into the code and set before the entry is saved in DB. If you need to change this you will need to extend the register function for now.
> 
> the behavior you describe doesnt match when user is created in admin. this seems inconsistent.

Registration and manually creating a user are two very different processes, manually creating a user doesn't trigger the confirmation email.

**Reply by hkovacs** [reply]
> > > Hello @hkovacs this is the intended behaviour :) To disable auto confirmation you cano enable the `email_confirmation` option.
> > > The default field won't be taken into account as the auto confirmation is built into the code and set before the entry is saved in DB. If you need to change this you will need to extend the register function for now.
> > 
> > 
> > the behavior you describe doesnt match when user is created in admin. this seems inconsistent.
> 
> Registration and manually creating a user are two very different processes, manually creating a user doesn't trigger the confirmation email.

is there a link to guide that discusses this?

**Comment by derrickmehaffy** [comment]
Moving this to a discussion as it's intended, but can continue to discuss.
